### PR TITLE
feat(zk): create halo2 proving key by deser

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR creates Halo2 Proving Key by deserializing the byte array which is serialized from https://github.com/kroma-network/halo2/commit/746e4ac62ea39e8fec9f711a6b5a359dfa596818.

`permutation_proving_key_` and `vanishing_argument_` of the `ProvingKey` are going to be built internally after `Prover` binding is completed.

### CODE REVIEW

**Comment by TomTaehoonKim** [comment]
@chokobole Is ? in [d7623d5](https://github.com/kroma-network/tachyon/pull/243/commits/d7623d57849fb272878b9dc255219390fe23b03c) typo?

**Comment by chokobole** [comment]
> @chokobole Is ? in [d7623d5](https://github.com/kroma-network/tachyon/pull/243/commits/d7623d57849fb272878b9dc255219390fe23b03c) typo?

I meant regular expressions. If it doesn't deliever clear message, then i can change the commit message!

**Comment by TomTaehoonKim** [comment]
@chokobole Nah, I think it's fine

File: vendors/halo2/src/lib.rs [hunk 1]
```diff
@@ -4,5 +4,6 @@ mod circuits;
 mod consts;
 mod msm;
 mod prover;
-mod xor_shift_rng;
+mod proving_key;
 mod transcript;
+mod xor_shift_rng;
```
Comment (TomTaehoonKim) [inline vendors/halo2/src/lib.rs:9]: :+1:

**Review by TomTaehoonKim — CHANGES_REQUESTED** [review]

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Comment by ashjeong** [comment]
https://github.com/kroma-network/tachyon/pull/243/commits/0f6f531686a202fa96a6a12e68724a2f17e3284a
Rather than a "refac" commit I believe this should be a "style" commit as per [Angular Convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines) and [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/); however, we do not currently use "style" type commits as per Kroma's [.github](https://github.com/kroma-network/.github/blob/main/CONTRIBUTING.md). Perhaps we should take this into account and discuss further later on.

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**